### PR TITLE
Improve `ui_combo_12` ui for custom color presets

### PR DESCRIPTION
**Pull request by crumblingstatue**
Shows color swatches for both the preset color and the resulting token

image

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -191,22 +191,45 @@ impl Colorix {
...
-                                self.update_color(ctx, i);
-                            };
+                ui.horizontal(|ui| {
+                    if let Some(ColorPreset::Custom(rgb)) = self.theme.get_mut(i) {
+                        let re = ui.color_edit_button_srgb(rgb);
+                        if re.changed() {
+                            self.update_color(ctx, i);
                         }
-                    });
+                        let tokens = [
```
Comment (crumblingstatue) [inline src/lib.rs]: This is kinda questionable, but I'm not sure how else to get the resulting tokens

File: src/tokens.rs [hunk 2]
```diff
@@ -214,4 +214,31 @@ impl ColorPreset {
             Self::Custom([r, g, b]) => [r, g, b],
         }
     }
+    pub(crate) const fn label(self) -> &'static str {
```
Comment (crumblingstatue) [inline src/tokens.rs:235]: I added this so I can make `Custom` label in the dropdown plain, without messing with the `Debug` impl.

File: src/lib.rs (hunk 1 above)
Comment (frankvgompel) [reply src/lib.rs]: Quite. I'll think about it.

File: src/tokens.rs (hunk 2 above)
Comment (frankvgompel) [reply src/tokens.rs:235]: Can we replace DROPDOWN_TEXT with `self.themes[i].label()`?

Comment (crumblingstatue) [reply src/tokens.rs:235]: > Can we replace DROPDOWN_TEXT with `self.themes[i].label()`?

Done

File: src/lib.rs (hunk 1 above)
Comment (crumblingstatue) [reply src/lib.rs]: > Quite. I'll think about it.

If you figure some API out, I can rebase my pull request on top of it

File: src/tokens.rs (hunk 2 above)
Comment (frankvgompel) [reply src/tokens.rs:235]: On 2nd thought: leave it as it was.

File: src/lib.rs (hunk 1 above)
Comment (frankvgompel) [reply src/lib.rs]: You can remove this and instead of `token[i]` you should be able to use the new `self.theme[i].get_token(i)`

Comment (crumblingstatue) [reply src/lib.rs]: Done

File: src/tokens.rs (hunk 2 above)
Comment (crumblingstatue) [reply src/tokens.rs:235]: > On 2nd thought: leave it as it was.

Can you explain? Do you want me to restore the `DROPDOWN_TEXT` array (that I removed in a commit)?
What's the rationale behind that?

**Comment by frankvgompel** [comment]
Abolutely brilliant work. Thanks.

**Pull request merged**